app/features-json/artifact_json_controller.rb

```diff
@@ -0,0 +1,79 @@
+require_relative "./json_authenticated_controller_base"
+require_relative "./view_models/artifact_view_model"
+
+module FastlaneCI
+  # Controller for providing all data relating to artifacts
+  class ArtifactJSONController < JSONAuthenticatedControllerBase
+    HOME = "/data/project/:project_id/build/:build_number/artifact"
+
+    class ArtifactNotFoundError < FastlaneCIError
```

Comment:
Lets use `halt(404, header, body)` instead of raise and catch. halting uses throw/catch so that we don't have to worry about proceeding further in the method.